simfection/interaction_engine.py

```diff
@@ -20,6 +28,7 @@ def __init__(self,
         self.pathogen = {
             key: settings.get_setting(key) for key in settings.get_setting('pathogen_keys')
         }
+        self.settings = settings
```

Comment:
We want to avoid carrying around duplicate copies of information. Rather than loading `settings` into the `interaction` unpack only the settings that the interaction needs.